### PR TITLE
[FIXED JENKINS-31915] Proxy settings in plugins page are ignored

### PR DESCRIPTION
**Pull request by escoem**
Added [NTLM support](http://hc.apache.org/httpclient-3.x/authentication.html#NTLM) to doValidateProxy and noProxyHost is never ignored.

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/ProxyConfiguration.java [hunk 1]
```diff
@@ -333,5 +342,26 @@ public FormValidation doValidateProxy(
...
+                }
+            }
+            return false;
+        }
+
+        private Credentials createCredentials(String userName, String password) {
+            if (userName.indexOf('\\') >= 0){
+                final String domain = userName.substring(0, userName.indexOf('\\'));
+                final String user = userName.substring(userName.indexOf('\\') + 1);
+                return new NTCredentials(user, Secret.fromString(password).getPlainText(), domain, "");
```
Comment (stephenc) [inline core/src/main/java/hudson/ProxyConfiguration.java]: what happens if you are connecting to a non-NTLM proxy where `\` is a valid character in the username?

**Comment by stephenc** [comment]
:bee: but I do wonder about the case where the proxy is using crazy characters in the username

**Comment by amuniz** [comment]
I agree that the '\' in usernames non-ntlm could be a problem. This could be fixed (without breaking compatibility with previous configurations) by adding a configuration option (a checkbox) to indicate if you are using NTLM auth or not (disabled by default), so `userName.indexOf('\\') >= 0` would be only checked if NTLM is active.

This is a :bee: from me, but I would encourage you to consider that option.

**Comment by escoem** [comment]
Please @stephenc , @amuniz , I have updated the help file removing obsolete references to setting configuration as system property.

**Comment by amuniz** [comment]
re-:bee:ing

**Comment by andresrc** [comment]
:bee:

Regarding the `\` in user names, I assume that, as we say in spanish, "when we reach that river, we will walk through that bridge" :wink:

**Comment by escoem** [comment]
Thanks! 

@reviewbybees done

**Comment by oleg-nenashev** [comment]
@escoem Merge conflict :(

**Comment by oleg-nenashev** [comment]
@escoem bump

**Comment by escoem** [comment]
hi @oleg-nenashev I have to review it and I will do asap. :-)

**Pull request merged**